lib/kay/src/compact_dict.rs

```diff
@@ -32,21 +41,25 @@ impl <K: Eq + Copy, V: Compact + Clone, A: Allocator> CompactDict<K, V, A> {
         None
     }
 
+    /// Lookup the item in the dict, but also swaps the item with the first element so lookup for
+    /// that item will be faster
     pub fn get_mru(&mut self, query: K) -> Option<&V> {
         for i in 0..self.keys.len() {
             if self.keys[i] == query {
-                self.keys.swap(0, i);
+                self.keys.swap(0, i); // TODO: Is is better to swap with the one in front instead of the first?
```

Comment:
Instead of promoting the item to the front of the list, promote it one level forwards